### ISSUE TITLE
Push to merge

### ISSUE THREAD

**Issue by Ignoble61** [Ignoble61/skin.aeon.madnox#3]
With #2 having reached such length that loading it is now causing my tablet to crash, can we please move discussion to this thread for what we still need to do to get the merge done.

From what I can see there are three things needed:-

- [x] Updates for script.skinshortcuts 1.0.7
- [x] Use Schimi's new language strings for widget options
- [x] Mr. V to finish current work on widget cases

As script.skinshortcuts has just been updated on the repo (though it will take a little time to get to all mirrors, and even longer to get to all systems) we can now go ahead with the first. I've updated the template.xml file as needed - Mr. V, are you still happy to make the other changes (I've added some extra background options that will need changing!)

Language strings is a slog, but I can work through those tomorrow unless anyone else wants to do it.

And after that, Mr. V, I think we're just waiting for the nod from you :)

**Comment by schimi2k** [comment]
Me and my tablet wanna thank you too for this :)

**Comment by bartolomesoriano** [comment]
@Ignoble61 script.skinshortcuts in my installs already shows it is at version 1.0.7.  Is there any additional action I need to take?

**Comment by Ignoble61** [comment]
If you've manually installed the git version at any point then you probably want to update to the latest git version to ensure you have all the fixes that are in this release. Otherwise, you're ready to go.

**Comment by bartolomesoriano** [comment]
Thanks Bob.  Done.

**Comment by mistervee** [comment]
@Ignoble61 Clean thread. Awesome. I though it was just me with the slow loading time of the page.

Cases are done. Nothing more to do unless someone can find any bugs. I also updated the management dialog for 1.0.7. I try and do some of the language strings today for the widgets.

One the language strings are done I am ok to PR this.

**Comment by Ignoble61** [comment]
@mistervee Great stuff. Let me know what language strings are still to do, if you find the time to get to them. I'll finish them off in the morning.

After that - unless anyone has any objections - I'll pull the final few commits in the skinshortcuts branch and squash them, then do the PR.

**Comment by mistervee** [comment]
@Ignoble61 looks like I am not going to get time today to update the language strings for the widgets.

**Comment by Ignoble61** [comment]
No problem. I'm just working on trying to get the skin shortcuts context menu extension to the repo, then I'll switch to it. Do you know if its just in the management dialog that they need doing?

**Comment by mistervee** [comment]
That and custom_1120_GeneralMainMenuSettings.xml and possibly the variables.xml. Although the variable might break stuff unless you add some more variables such as
`StringCompare(Window.Property(xxxxxxxxxxxxx),somevalue)`
in things like
`<variable name="WidgetCaseLabelVar">
		<value condition="!IsEmpty(Window.Property(widgetCase))">$INFO[Window.Property(widgetCase)]</value>
		<value>$LOCALIZE[231]</value>
	</variable>`

**Comment by Ignoble61** [comment]
You're displaying values like the case name within the template...? The underlying value of the options isn't changing, it's only when we need to display that value to the user in a localised string that we need to worry. Either way, I'll check where everything I update is being used to ensure any conditions are correct wherever they're needed :)

**Comment by mistervee** [comment]
@Ignoble61 Not showing values to the user within the template just using the menu property's (template only property's). The custom_1120_GeneralMainMenuSettings.xml might me a bit different as I use skin strings for these and they also show the label to the user in some cases using the same skin strings. How you code it is all good, I have complete confidence in what ever you do.

**Comment by Ignoble61** [comment]
If you're just using the properties there then no changes needed - the properties themselves aren't affected by any of this. 1120 will need a little work, but should be easy enough.

> I have complete confidence in what ever you do.

Are you available for life coaching?

**Comment by mistervee** [comment]
> Are you available for life coaching?

You couldn't afford me.

**Comment by Ignoble61** [comment]
I'd question that, but I don't have the self-confidence to do so ;)

**Comment by Ignoble61** [comment]
@schimi2k - Found some untranslated strings for widget art choices, though if they're like the BD/DVD for the live backgrounds and aren't meant to be translated please ignore me :)

- Thumb
- ClearLogo
- DiscArt
- CharacterArt

@all - The widget option labels have all been updated to use localised strings now within the management dialog. However, typo's are a possibility so if you notice any missing or incorrectly translated labels, let me know. Now, break for lunch and then on to 1120...

**Comment by schimi2k** [comment]
Thx so much guys... I will check when my support ppl are done for today... And yes, these words shouldnt be translated bob :)

**Comment by Ignoble61** [comment]
Cheers Schimi. Unfortunately, from 1120, I'm pretty sure these two should be :wink: :-

* Match widget panel
* Always On

Does it cause any problems for the translations if I go ahead and add them in the english strings.po?

**Comment by MadMikeDoc** [comment]
@Ignoble61, Bob I'll do the infopanel alignement, think it's a slide issue, if you have done it already?

**Comment by bartolomesoriano** [comment]
@Ignoble61 I don't think adding those to the English strings.po should cause an issue.  Please let me know if you do, so I can update the Spanish accordingly.  Thanks.

I will review today every screen that I can in Spanish to ensure that my translations don't cause overlaps or cut-off text anywhere.

**Comment by Ignoble61** [comment]
@MadMikeDoc I added a slide animation to 1112 to move it up slightly if control 1827 is visible. I haven't touched the info panels themselves.

@bartolomesoriano Fair enough, and if Schimi complains on your head be it ;) I'll add the extra two strings and finish updating 1120 shortly.

**Comment by MadMikeDoc** [comment]
@Ignoble61 ok, well i pushed some changes, if they have made things worse we can revert them?

**Comment by Ignoble61** [comment]
@MadMikeDoc From a quick test 1112 is - unsurprisingly - slightly higher up, but it still functions perfectly. No reason to revert if you're happy with how they look.

**Comment by Ignoble61** [comment]
Ok, I've checked everything I can easily check and things seem to be working. It would be nice to get translations for the two new strings in from Schimi and Bart for German and Spanish respectively.

After that and aside from any final testing people want to do, is there anything we've missed? Or can we PR this when @MadMikeDoc @schimi2k and @bartolomesoriano have given it a final test run?

**Comment by MadMikeDoc** [comment]
@all it's good to go from me, can't find anything at the moment. Still testing though.

**Comment by bartolomesoriano** [comment]
@Ignoble61, @schimi2k here is the updated Spanish strings.po: 
[Spanish.zip](https://github.com/Ignoble61/skin.aeon.madnox/files/149420/Spanish.zip)

There are two strings that I did not see are personalized in the SSS management dialog: Choose item for menu and Restore Menu Items.

I also could not find the string "Mixed" (for the widget type) so I can translate it properly to Spanish (current translation is not the right one).

Now I will do a final round of testing.

**Comment by schimi2k** [comment]
sorry guys, i have some badass stress here :( realy busy, but i try to make a quick check now
@bartolomesoriano 
Mixed is default translation #20395
dont know about the other yet...

**Comment by Ignoble61** [comment]
Choose item for menu and Restore menu items are both strings provided by the Skin Shortcuts script. There's some changes going on in how addons and skins on the repo use Transifex - I'm hoping once they're sorted, we'll get the script on there and start to get translations for it again.

'Mixed' comes from Kodi's own strings - string 20395, as used in the smart playlist editor.

Thanks for the strings.po, but - and I know it's probably a daft question - it's to replace the one in the 'Spanish' folder...? I ask only because it's showing so many changes, and I don't want to go overwriting the wrong one!

**Comment by Ignoble61** [comment]
@schimi2k No hurry sir. Take your time and de-stress if you can :)

**Comment by bartolomesoriano** [comment]
@Ignoble61 yes it is to replace that one.  Today I also looked at every single string in that file and changed all the translations that were not correct, instead of just adding the two missing strings.

I did testing of all the functions I use, went through all the options to configure menus and submenus, changed a lot of the widget options such as cases, backgrounds, colors, panel info, etc.,
played music videos, movies, tv shows, and I have not found any issues.  From my perspective, we're good to go!

**Comment by Ignoble61** [comment]
Cheers, merged it in.

**Comment by bartolomesoriano** [comment]
@all one question, and pardon my ignorance.  If I change all the translated strings in Spanish to use properly accented vowels (á, é, etc.), does that create any problems? I ask because I saw that there were no accented vowels in the Spanish strings.po, and wasn't sure if I should correct that.

**Comment by MadMikeDoc** [comment]
@all, grh argh err! Since going to 16 release I seem to be loosing window backgrounds far more often, or my setup is corrupt somehow, frustrating when trying to check everthing thats all! ;-)

@bartolomesoriano, no idea but due your changes show up for you?

**Comment by schimi2k** [comment]
i found some... give me some minutes to verify ;)

**Comment by Ignoble61** [comment]
@bartolomesoriano There are plenty of accented characters in the other language files, so I can't believe it creates any problems.

@MadMikeDoc That could be the showstopper ;) (though I've no idea what we can actually do about it!) Have you tried running with a Textures.xbt?

**Comment by MadMikeDoc** [comment]
@Ignoble61 , no but I think it might just be me as no one else has mentioned it?

**Comment by bartolomesoriano** [comment]
@MadMikeDoc if you mean the translations I made they do show up for me.

@Ignoble61 thanks! I will then do proper translations and submit a new file in about an hour.

**Comment by Ignoble61** [comment]
@MadMikeDoc It's something I occasionally see - and not just with MadNox. I always put it down to too much testing, and a quick skin reload fixes it up.

**Comment by schimi2k** [comment]
#31823 was already in use Bob ;) thats why i add the list with the in use numbers hehe but this is something i can fix - so bart maybe wait a bit please
i also have some headings still in english ( seems i dont have them added on list...need to check this too )

**Comment by Ignoble61** [comment]
:blush: My apologies, I obviously didn't search hard enough when checking the strings were empty!

**Comment by schimi2k** [comment]
thats my GREAT AWESOME List for, that i add in language folder 
changed and free IDs.txt ... because the language file is a mess... maybe one day i can clean it up 
and no need to apologies ...

**Comment by bartolomesoriano** [comment]
@schimi2k No worries, just let me know when I can start.

**Comment by schimi2k** [comment]
damn, i am going locco... i changed the id from 31823 to 31821 - its free and i can see it in selection screen - when i select the ALWAYS ON option i get still the old " WORLDMAP AND ICON " text on settings window ?!?!?!? 
there was just the one 31823 number on 1120 ... any ideas ?

**Comment by Ignoble61** [comment]
variables.xml, line 4684 ;)

**Comment by schimi2k** [comment]
why is this all so complicated ? ... why cant we make this whole crap with photoshop :(
and sir bob, is it possible to id the header too ? 
<onclick>RunScript(script.toolbox,info=selectdialog,header=Show list widget info)</onclick>
Show list widget info in this case... there are some i think ... will add them to the english language file and pr it ?!

**Comment by schimi2k** [comment]
and can someone double check me please - try to refresh ( F5 ) madnox when you are on 1120 ... i get always a crash
( general options )

**Comment by Ignoble61** [comment]
I've no idea! Try changing it to header=$LOCALIZE[xxxxx] and see if it works - script.toolbox is written by phil65, so it's odds-on it will.

(And I think you meant "add them to the english language file, update the german language file and give bart time to update the spanish language file ... and pr it !?")

I can semi-confirm it, in that I was getting the same earlier when changing the strings in the first place. I can't re-confirm or diagnose because I put it down to my corrupt video database, which I'm currently 90% of the way through re-importing.

**Comment by schimi2k** [comment]
didnt you say some days ago we shouldnt use this $localize voodoo anymore ?

**Comment by MadMikeDoc** [comment]
@schimi2k , that was me, when in a <label> you don't need to use $LOCALIZE if that is all there is in the <label> for things like Bob is talking about you'll still need to use $LOCALIZE, at least I think you will?

**Comment by schimi2k** [comment]
hmmm yeah, bobs tip seems to work... not much longer now...

**Comment by schimi2k** [comment]
one small fix on 1112... but i will push the language stuff first

**Comment by schimi2k** [comment]
DONE
i have some texts that are too long in german... but we can fix this for RC-2 :)

i realy wish we could sit together in a bar, drink some Milk or a beer and just talk... about the weather or so... normal stuff...

**Comment by schimi2k** [comment]
@bartolomesoriano your final turn

**Comment by Ignoble61** [comment]
@schimi2k see my comment on your commit - you probably have taken it into account, but I can't just confirm till my library is back up. And it would be a bugger to break something that we only fixed yesterday ;)

**Comment by schimi2k** [comment]
no master jedi :) its just the GLOW behind it ;) it was too long and cut off at the bottom

**Comment by MadMikeDoc** [comment]
opps!

**Comment by schimi2k** [comment]
thats why i call it tiny cosmetic... if it would be something different, i would ask you before - i noticed that you are working on it ;)

**Comment by schimi2k** [comment]
opps ?

**Comment by Ignoble61** [comment]
No worries, just wanted to make sure :) (I'm at 97% re-importing, so I'll be able to actually look myself again soon!)

**Comment by schimi2k** [comment]
oha... grandpa´s final check ... i need to pee for an hour now... but i dont wanna miss the big moment hehe

**Comment by Ignoble61** [comment]
You need to pee for an hour, and you're calling me old! :P

**Comment by schimi2k** [comment]
but i can hold it :P

**Comment by schimi2k** [comment]
damn, but not much longer ...

**Comment by Ignoble61** [comment]
<realisation>Ah, you've needed to pee for an hour, not that you'll be peeing for an hour</realisation> You've got time. Bart said it would take him an hour to do the translations, and thats if he gets right on it. And I promise not to do the PR till you get back ;)

**Comment by schimi2k** [comment]
pew back
one other thing you can do this weekend maybe Sir Bob ... can you show DOC how he can auto activate the commits from us ? before i can PR my updates to DOC, but we have to wait before he "accept" it - here on this dev version i can commit direct from my tool to the git... this would be cool if he can give us permission on master git too ? ( me, meowmoo,Bob,V and SIlvo maybe )

**Comment by Ignoble61** [comment]
I think he might have worked it out, as I got a notification that I've been added to MadMikeDoc/skin.aeon.madnox ... but Settings, at the top of the github page, then Collaborators in the left hand menu.

**Comment by schimi2k** [comment]
:+1:

**Comment by MadMikeDoc** [comment]
Yes, was in the process of adding you as collaborator. Added Sir Bob just in case this merge is a bit much for me and his wisdom and knowledge could help.

**Comment by schimi2k** [comment]
now i get a message too... after everybody else ... PHA ;)

**Comment by schimi2k** [comment]
i would be more then happy if we can still keep in contact... not only because of all this work... i realy like this old man ;)

**Comment by Ignoble61** [comment]
It's going to be as easy as clicking the big green button, Mike :)

I have a feeling (based on experience with helping other skins to integrate) there'll be one or two thousand skin shortcuts related questions on the forums, so you'll definitely still see me around for quite a while! And next time I'm in Germany I am looking you up for some of your famous milk.

**Comment by schimi2k** [comment]
i try to make some general settings tipps tomorrow and then wait for all the complainings and maybe questions, and see, what else i can add to my tipps... Bad time for our Bet...on weekend they have more time to test all the new stuff...

**Comment by schimi2k** [comment]
Lactose free ! or how i call it " FART-FREE "

**Comment by MadMikeDoc** [comment]
@Ignoble61 , Bob does the version numbers in the addon.xml matter? This is what we have and I've never updated the version part, is it something I should do now?

`<import addon="xbmc.gui" version="5.10.0"/>
    <import addon="script.skinshortcuts" version="1.0.7"/>
    <import addon="script.skin.helper.service" version="1.0.0"/>
    <import addon="service.library.data.provider" version="0.0.7"/>
    <import addon="script.toolbox" version="1.1.0"/>
    <import addon="script.image.resource.select" version="0.0.5"/>`

**Comment by Ignoble61** [comment]
The xbmc.gui part is important, as it specifies which version of Kodi the skin actually works with - but that's set to Jarvis, so looks alright.

The rest specify which addons your skin relies on, and the minimum version number required. When installing via zip, Kodi will install any that aren't present (or refuse to install the skin). The only version number that is important there, I suspect, is the skin shortcuts one as you do require 1.0.7. In truth, though, users will have the most up-to-date version of them all because they're all available on the repo.

**Comment by MadMikeDoc** [comment]
Thanks Bob, I'll just update the version number for Madnox then in addon.xml

**Comment by Ignoble61** [comment]
@MadMikeDoc Kodi convention is to use a 3-part version number - x.x.x (I don't know if you have repo ambitions, but it's absolutely required there) :)

**Comment by MadMikeDoc** [comment]
so is 0.9.0 ok?

**Comment by schimi2k** [comment]
So we can rls 9 rcs ;)

**Comment by Ignoble61** [comment]
It's absolutely fine, if it's got those three digits :) (And there's nothing strictly wrong with 0.9, it's just convention!)

**Comment by MadMikeDoc** [comment]
Thanks Sir Bob of Bobbyness.

**Comment by MadMikeDoc** [comment]
we keep being asked for a repo but understand you have to be below 100 MB in size, which isn't going to happen. I've tried twice tonight using the texture tool to create a textures.xbt from the media folder and both times the texture tool has crashed on Windows 10. Madnox is so big it even kills the texture tool!

**Comment by mistervee** [comment]
Morning All,

Seems like missed out on all the fun the language strings.

Am I too late to do a quick fix?

**Comment by Ignoble61** [comment]
I have one built on Mac here if you want it - it's built against the Krypton texturepacker, but works fine against Jarvis for me. The only problem is my very poor internet speeds - at 206mb it will take a long time to upload.

**Comment by Ignoble61** [comment]
@mistervee - Not at all. Anything is fair game until @bartolomesoriano's spanish Strings.po :)

**Comment by MadMikeDoc** [comment]
I'll try again after my trip so we'll see then if I can get texture tool to work or not, but thanks Bob for the offer.

**Comment by bartolomesoriano** [comment]
@all sorry I was away for a bit.  I will get right on this!

**Comment by mistervee** [comment]
Hopefully I have fix the last of the bugs.

**Comment by Ignoble61** [comment]
@bartolomesoriano There's no hurry - we, and all MadNox's Spanish users, appreciate the time you can spare :)

**Comment by bartolomesoriano** [comment]
@all here is the updated Spanish strings.po: 
[Spanish.zip](https://github.com/Ignoble61/skin.aeon.madnox/files/149573/Spanish.zip)

**Comment by mistervee** [comment]
@bartolomesoriano Committed to Git. Thanks for your assistance.

**Comment by Ignoble61** [comment]
Thanks to everyone - I think we did it :)

**Comment by mistervee** [comment]
Great Job by all. A big thanks to everyone for there contributions.

**Comment by bartolomesoriano** [comment]
Woohoo!! Thank you all, feels good to be in this moment! :D

**Comment by MadMikeDoc** [comment]
and merged it is. Quite outstanding what has been done and again eternal thanks to the A-Team for without you it simply would not have happened. Many thanks @bartolomesoriano for the tireless testing and translating and to team Madnox for all our contributions to this project.

**Comment by Ignoble61** [comment]
Mike, thanks for letting us do it! I imagine its been at least a little traumatic seeing us break your skin ;)

On a serious note, I'm about to delete the skinshortcuts and skinshortcuts-dev branches. I'll happily leave this fork up, though, if anyone wants to use it as a place to do fixes before PR'ing them back.

**Comment by MadMikeDoc** [comment]
Perhaps leave -dev up at least for a while? I think a dev branch is something that could be useful to Madnox in the future.

**Comment by Ignoble61** [comment]
Dev branch now won't merge into your master without rebasing, so it's really not a good place for further work - and for various reasons it's history is messed up so it's not easy to rebase. I've just updated the master on the fork to be equal to your master, and a new dev branch can be created off of that if people find that useful.

**Comment by bartolomesoriano** [comment]
@all, one small (but not unimportant) thing.  i switched to @MadMikeDoc 's master branch, and when I start Kodi I don't have @schimi2k 's intro video anymore!

**Comment by Ignoble61** [comment]
How on Earth did that happen??? I'll PR it in a minute or two.

**Comment by bartolomesoriano** [comment]
Thanks Bob, before we start to hear swearing in German!

**Comment by Ignoble61** [comment]
All I can think is a commit somehow got dropped whilst it was all being squashed, so I'm going to just do a check that there's no other files that should have changed that haven't :)

**Comment by schimi2k** [comment]
... One happy and one sad side now... I am realy happy what we did here... But i will miss all this crazy stuff - but its realy great to know you freaks now :) you guys are awesome!  Just a huge THX

**Comment by MadMikeDoc** [comment]
@all an open invitation from me that if you ever need help with something and you think I / we from team Madnox  can help then just say the word, will always be a pleasure to work with you again in some capacity and if anyone ever wants to pop in and help team Madnox then please do, but not anytime soon as you all need a break and you all have you own things to take care of.

Thanks Bob for checking the commit

**Comment by bartolomesoriano** [comment]
Thanks Mike! You can  be sure I will continue checking the forum and the development of the most awesome skin for Kodi! :smile: 

Just out of curiosity, where are you all located? I know @schimi2k is in Germany, and I am in the Dominican Republic.

**Comment by MadMikeDoc** [comment]
I'm in the U.K. @bartolomesoriano . We have had one fantastic international team here from the looks of it.

**Comment by Ignoble61** [comment]
Phew, just the intro video that got missed :) Just PR'd it. (Another from the UK here)

**Comment by schimi2k** [comment]
I wanna agree with doc... However i can help with ANY gfx stuff... Anytime :)

**Comment by schimi2k** [comment]
Uk-Australia-domrep-germany and is silvo from the us?

**Comment by MadMikeDoc** [comment]
Bob, what part of the Uk?. I think Mr. Silvo might be from Spain, not sure why but maybe that area?

**Comment by Ignoble61** [comment]
I'm somewhere near Manchester.

**Comment by MadMikeDoc** [comment]
Somewhere near Birmingham.

**Comment by mikesilvo164** [comment]
Wow, Christmas 2016 came early this year :stuck_out_tongue_winking_eye: 

@MadMikeDoc I'm a New Yorker born and bred. Long Island specifically. Now its time to save up and come visit you guys :smiley:

**Comment by bartolomesoriano** [comment]
Quite the international team, cool! You are all more than welcome to come enjoy our eternal summer here in the Dominican Republic! If you ever plan on a vacation here, please let me know!

**Comment by schimi2k** [comment]
DomRep... its famouse here for the nice girls ;)
